esphome/components/tuya/tuya.cpp

```diff
@@ -22,6 +22,7 @@ void Tuya::loop() {
 
 void Tuya::dump_config() {
   ESP_LOGCONFIG(TAG, "Tuya:");
+  ESP_LOGCONFIG(TAG, "  Product: %s", this->product_.c_str());
```

Comment:
```suggestion
  ESP_LOGCONFIG(TAG, "  Product: '%s'", this->product_.c_str());
```

If the component is not receiving anything, the log output is not ideal